pkg/lib/service.js

```diff
@@ -292,7 +292,20 @@
 
         function call_manager_with_reload(method, args) {
             return call_manager(method, args).then(function () {
-                return call_manager("Reload", [ ]);
+                var dfd = cockpit.defer();
+                call_manager("Reload", [ ]).
+                    done(function () { dfd.resolve(); }).
+                    fail(function (error) {
+                        // HACK: https://bugzilla.redhat.com/show_bug.cgi?id=1560549
+                        // some systemd versions disconnect too fast from the bus
+                        if (error.name === "org.freedesktop.DBus.Error.NoReply") {
+                            refresh();
```

Comment:
This is here because we also don't get the `Reloading(false)` signal, right?

Reply:
Correct, otherwise the UI never updates by itself. I had that without the refresh first, and it took a manual `systemctl daemon-reload` or a page reload. With the refresh it works, and the test succeeds (`check-services TestServices.testApi` in #8874)